packages/components/src/CodeEditor.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC } from "react";
+import AceEditor from "react-ace";
+
+import "ace-builds/src-noconflict/mode-golang";
+import "ace-builds/src-noconflict/theme-github";
+import "ace-builds/src-noconflict/ext-language_tools";
+import "ace-builds/src-noconflict/keybinding-vim";
+
+interface CodeEditorProps {
+  value: string;
+  onChange: (value: string) => void;
+  oneLine: boolean;
+  width?: number;
+}
+
+export let CodeEditor: FC<CodeEditorProps> = (props) => (
+  <AceEditor
+    value={props.value}
+    mode="golang"
+    theme="github"
+    width={props.width ? props.width + "px" : "500px"}
+    height={props.oneLine ? "1.2em" : "500px"}
```

Comment:
I recommend trying to change the height of one line; it's currently a bit buggy for me.